src/search/pdbs/pattern_collection_generator_multiple.cc

```diff
@@ -0,0 +1,310 @@
+
+        int remaining_pdb_size = min(remaining_collection_size, max_pdb_size);
+        double remaining_time =
+            min(static_cast<double>(timer.get_remaining_time()), pattern_generation_max_time);
+
+        PatternInformation pattern_info = compute_pattern(
+            remaining_pdb_size,
+            remaining_time,
+            pattern_computation_rng,
+            task,
+            goals[goal_index],
+            move(blacklisted_variables));
+        handle_generated_pattern(
+            move(pattern_info),
+            generated_patterns,
+            generated_pdbs,
+            timer);
+
+        if (collection_size_limit_reached() ||
+            time_limit_reached(timer) ||
+            check_for_stagnation(timer)) {
+            break;
+        }
+
+        ++num_iterations;
+        ++goal_index;
+        goal_index = goal_index % goals.size();
+        assert(utils::in_bounds(goal_index, goals));
+    }
+
+    PatternCollectionInformation result = get_pattern_collection(task_proxy, generated_pdbs);
+    if (verbosity >= utils::Verbosity::NORMAL) {
+        utils::g_log << get_name() << " number of iterations: "
+                     << num_iterations << endl;
+        utils::g_log << get_name() << " average time per generator: "
+                     << timer.get_elapsed_time() / num_iterations
+                     << endl;
+        dump_pattern_collection_generation_statistics(
+            get_name(),
+            timer.get_elapsed_time(),
+            result);
+    }
+    return result;
+}
+
+void add_multiple_options_to_parser(options::OptionParser &parser) {
+    parser.add_option<int>(
+        "max_pdb_size",
+        "maximum number of states for each pattern database, computed "
+        "by compute_pattern (possibly ignored by singleton patterns consisting "
+        "of a goal variable)",
+        "2000000",
+        Bounds("1", "infinity"));
+    parser.add_option<int>(
+        "max_collection_size",
+        "maximum number of states in all pattern databases of the "
+        "collection (possibly ignored, see max_pdb_size)",
+        "20000000",
+        Bounds("1", "infinity"));
+    parser.add_option<double>(
+        "pattern_generation_max_time",
+        "maximum time in seconds for each call to the algorithm for "
+        "computing a single pattern",
+        "infinity",
+        Bounds("0.0", "infinity"));
+    parser.add_option<double>(
+        "total_max_time",
+        "maximum time in seconds for this pattern collection generator. The "
+        "algorithm will always execute at least one iteration, i.e., call the "
+        "algorithm for computing a single pattern at least once.",
+        "100.0",
+        Bounds("0.0", "infinity"));
+    parser.add_option<double>(
+        "stagnation_limit",
+        "maximum time in seconds the algorithm allows without generating a "
+        "new pattern. The multiple algorithm terminates prematurely if this "
```

Comment:
"The multiple algorithm" sounds strange, especially without context. Maybe just replace this with "Pattern generation terminates ...".